### PR TITLE
Make FS return a redirect for directories without trailing slash

### LINKED ISSUE #792: FS: redirect directories to have a trailing slash
Currently, I can access a file `/test/index.html` either through `/test/` or through `/test`. In the latter case, relative paths in the document will refer to completely different files, which makes this unappropriate in my opinion - `/test` should yield a 301 redirect to `/test/`. This will be a breaking change if someone uses something like `ctx.Request.SetRequestURI(path.Clean(p))` before calling the FSHandler though, so I guess making it an option would make sense?.

### PR DESCRIPTION
**Pull request by moqmar**
Fixes #792

### CODE REVIEW

**Comment by erikdubbelboer** [comment]
Thanks. Can you add a test case please?

**Comment by moqmar** [comment]
I've added a test case and actually found an issue due to that :D

I'm not completely sure though if there are other functions making use of `fs` though that way (overriding the request URI before letting it serve a file or directory) - I haven't found any yet at least.

File: fs_test.go [hunk 1]
```diff
@@ -738,3 +738,33 @@ func TestServeFileContentType(t *testing.T) {
...
+	ServeFile(&ctx, ".git/")
+	if ctx.Response.StatusCode() != StatusOK {
+		t.Fatalf("Unexpected status code %d for directory '/.git/' with trailing slash. Expecting %d.", ctx.Response.StatusCode(), StatusOK)
+	}
+
+	ctx.Request.Reset()
+	ctx.Response.Reset()
+	ServeFile(&ctx, "fs.go")
+	if ctx.Response.StatusCode() != StatusOK {
+		t.Fatalf("Unexpected status code %d for file '/.git'. Expecting %d.", ctx.Response.StatusCode(), StatusOK)
```
Comment (erikdubbelboer) [inline fs_test.go:768]:
```suggestion
		t.Fatalf("Unexpected status code %d for file '/fs.go'. Expecting %d.", ctx.Response.StatusCode(), StatusOK)
```

**Comment by kimkit** [comment]
@moqmar is it better to use origin path for redirect ?

https://github.com/valyala/fasthttp/blob/master/fs.go#L738

```go
ctx.RedirectBytes(append(ctx.Path(), '/'), StatusFound)
```

**Pull request merged**